src/features/lightspeed/utils/oneClickTrial.ts

```diff
@@ -3,23 +3,43 @@ import { IError } from "@ansible/ansible-language-server/src/interfaces/lightspe
 import { lightSpeedManager } from "../../../extension";
 import { LightSpeedCommands } from "../../../definitions/lightspeed";
 
-export async function showTrialInfoPopup(error?: IError): Promise<boolean> {
-  if (
-    lightSpeedManager.lightspeedExplorerProvider
-      .lightspeedExperimentalEnabled &&
-    error?.code === "permission_denied__user_has_no_subscription"
-  ) {
-    const buttonLabel = "Start my Trial period, It's free!";
-    const selection = await vscode.window.showInformationMessage(
-      "Oh! You don't have an active Lightspeed Subscription",
-      buttonLabel,
-    );
-    if (selection === buttonLabel) {
-      vscode.commands.executeCommand(
-        LightSpeedCommands.LIGHTSPEED_OPEN_TRIAL_PAGE,
+export class OneClickTrialProvider {
+  public mapError(mappedError: IError): IError {
+    // Don't show "Apply for Trial" message if experimental features are not enabled
+    if (
+      mappedError?.code === "permission_denied__can_apply_for_trial" &&
+      !lightSpeedManager.lightspeedExplorerProvider
```

Comment:
We don't need the experimental check.

Reply:
Ok. We can keep it for now, but since the feature should be enabled server side first and we want the clients to be ready as quickly as possible.

BTW, we must also avoid pushing a release of this without having a fully functional `permission_denied__can_apply_for_trial` support behind.